### ISSUE TITLE
feature request: add support for the --resolve option in curl plugin (CURLOPT_RESOLVE)?

### ISSUE THREAD

**Issue by bpkroth** [collectd/collectd#2171]
Hi, using collectd 5.6 and 5.7 on Debian.

We have a number of websites where I need to check IPv4 and IPv6 paths separately.

In some cases we're able to use the IP address in the URL attribute of the configuration and then send an appropriate Host value in a Header attribute.

Here's what an example conf might look like:

```
LoadPlugin "curl"
<Plugin "curl">
        <Page "https://f.q.d.n/">
                URL "https://10.1.2.3/"
                Header "Host: f.q.d.n"
                Header "Cache-Control: max-age=0"
                MeasureResponseTime true
                VerifyPeer false
                VerifyHost false
        </Page>
        <Page "https-6://f.q.d.n/">
                URL "https://fded::1234/"
                Header "Host: f.q.d.n"
                Header "Cache-Control: max-age=0"
                MeasureResponseTime true
                VerifyPeer false
                VerifyHost false
        </Page>
</Plugin>
```

The trouble is that this fails in strange ways when SNI support and/or redirects to different hostnames are involved (eg: `http://f.q.d.n/old/path` redirects to `https://new.f.q.d.n/new/path` for some or all paths being monitored).  For instance, the curl plugin continues to send the original Host attribute requested in the configuration (`f.q.d.n` in this case) which breaks SNI if the redirected hostname changes (note that the cli `curl` does not do this), or else the redirected hostname followed no longer uses the original IP requested (eg: switches from IPv4 to IPv6 or vise-versa).

Some combination of telling curl how to resolve hostnames to IPs for that series of requests (eg: via `CURLOPT_RESOLVE`) and/or adjusting the redirect behavior to optionally only allow following redirects within the same host would help solve this particular issue for us.

I haven't had a chance to look through the code much yet to see how feasible this is or get started on a patch myself yet, but I figured I'd post the request first and see if anyone had any other thoughts on the matter first.

Thanks,
Brian

**Comment by shastah** [comment]
@bpkroth, I needed something similar and came up with #2986 - have a look at it, maybe it'll be useful in your case too.

**Comment by bpkroth** [comment]
Sorry for the delay.  Thanks for reaching out.  I don't work on the project that had that particular issue anymore so I can't test it directly, but I'll pass it along in case it's still relevant (@uwkhahn).  At a quick skim it looked reasonable.  Thanks!

**Comment by rpv-tomsk** [comment]
Solved in  #2986.
Thanks for using Collectd.
